hikari/impl/event_manager.py

```diff
@@ -113,82 +136,168 @@ async def on_channel_delete(self, shard: gateway_shard.GatewayShard, payload: da
 
         await self.dispatch(event)
 
+    @event_manager_base.filtered((channel_events.GuildPinsUpdateEvent, channel_events.DMPinsUpdateEvent))
     async def on_channel_pins_update(self, shard: gateway_shard.GatewayShard, payload: data_binding.JSONObject) -> None:
         """See https://discord.com/developers/docs/topics/gateway#channel-pins-update for more info."""
         # TODO: we need a method for this specifically
         await self.dispatch(self._app.event_factory.deserialize_channel_pins_update_event(shard, payload))
 
+    # Internal granularity is preferred for GUILD_CREATE over decorator based filtering due to its large cache scope.
     async def on_guild_create(self, shard: gateway_shard.GatewayShard, payload: data_binding.JSONObject) -> None:
```

Comment:
These could actually hold the `filtered` metadata with their correct settings. And just add a comment saying that the function will later decrease the scope of what is cached + deserialized based on the information

Reply:
Seems unnecessary to duplicate steps like that for every guild create and update event dispatch